docs/custom_promise_types/rss.py

```diff
@@ -0,0 +1,172 @@
+import requests, html, re, os, random
+import xml.etree.ElementTree as ET
+from cfengine import PromiseModule, ValidationError, Result
+
+
+class RssPromiseTypeModule(PromiseModule):
+    def __init__(self):
+        super().__init__("rss_promise_module", "0.0.2")
+
+
+    def validate_promise(self, promiser, attributes):
+        # check promiser type
+        if type(promiser) is not str:
+            raise ValidationError("invalid type for promiser: expected string")
+
+        # check that promiser is a valid file path
+        if not self._is_unix_file(promiser) and not self._is_win_file(promiser):
+            raise ValidationError(f"invalid value '{promiser}' for promiser: must be a filepath")
+
+        # check that required attribute feed is present
+        if "feed" not in attributes:
+            raise ValidationError("Missing required attribute feed")
+
+        # check that attribute feed has a valid type
+        feed = attributes['feed']
+        if type(feed) is not str:
+            raise ValidationError("Invalid type for attribute feed: expected string")
+
+        # check that attribute feed is a valid file path or url
+        if not (self._is_unix_file(feed) or self._is_win_file(feed) or self._is_url(feed)):
+            raise ValidationError(f"Invalid value '{feed}' for attribute feed: must be a file path or url")
+
+        # additional checks if optional attribute select is present
+        if "select" in attributes:
+            select = attributes['select']
+
+            # check that attribute select has a valid type
+            if type(select) is not str:
+                raise ValidationError(f"Invalid type for attribute select: expected string")
+
+            # check that attribute select has a valid value
+            if select != 'newest' and select != 'oldest' and select != 'random':
+                raise ValidationError(f"Invalid value '{select}' for attribute select: must be newest, oldest or random")
+
+
+    def evaluate_promise(self, promiser, attributes):
+        # get attriute feed
+        feed = attributes['feed']
+
+        # fetch resource
+        resource = self._get_resource(feed)
+        if (resource == None):
```

Comment:
`is not None` is a more common way for comparing values with `None` in Python. (`None` is a singleton)

Reply:
Also, no parentheses needed around the condition.